ngmix/ksigmamom.py

```diff
@@ -0,0 +1,445 @@
+            Defaults to False.
+        no_psf : bool, optional
+            If True, allow inputs without a PSF observation. Defaults to False
+            so that any input observation without a PSF will raise an error.
+
+        Returns
+        -------
+        result dictionary
+        """
+        if not isinstance(obs, Observation):
+            raise ValueError("input obs must be an Observation")
+
+        if not obs.has_psf() and not no_psf:
+            raise RuntimeError("The PSF must be set to measure a pre-PSF moment!")
+
+        if not no_psf:
+            psf_obs = obs.get_psf()
+
+            if psf_obs.jacobian.get_galsim_wcs() != obs.jacobian.get_galsim_wcs():
+                raise RuntimeError(
+                    "The PSF and observation must have the same WCS "
+                    "Jacobian for measuring pre-PSF moments."
+                )
+        else:
+            psf_obs = None
+
+        return self._meas_fourier_only(obs, psf_obs, return_kernels)
+
+    def _meas_fourier_only(self, obs, psf_obs, return_kernels):
+        # pick the larger size
+        if psf_obs is not None:
+            if obs.image.shape[0] > psf_obs.image.shape[0]:
+                target_dim = int(obs.image.shape[0] * self.pad_factor)
+            else:
+                target_dim = int(psf_obs.image.shape[0] * self.pad_factor)
+        else:
+            target_dim = int(obs.image.shape[0] * self.pad_factor)
+        eff_pad_factor = target_dim / obs.image.shape[0]
+
+        # pad image, psf and weight map, get FFTs, apply cen_phases
+        kim, im_row, im_col = _zero_pad_and_compute_fft(
+            obs.image, obs.jacobian.row0, obs.jacobian.col0, target_dim,
+        )
+        fft_dim = kim.shape[0]
+
+        if psf_obs is not None:
+            kpsf_im, psf_im_row, psf_im_col = _zero_pad_and_compute_fft(
+                psf_obs.image,
+                psf_obs.jacobian.row0, psf_obs.jacobian.col0,
+                target_dim,
+            )
+        else:
+            # delta function in k-space
+            kpsf_im = np.ones_like(kim, dtype=np.complex128)
+            psf_im_row = 0.0
+            psf_im_col = 0.0
+
+        # the final, deconvolved image we want is
+        #
+        #  deconv_im = kim * im_cen_phase / (kpsf_im * psf_imcen_phase)
+        #
+        # For efficiency we combine the phase comps to reduce sin and cos calls
+        # like this
+        #
+        #  deconv_im = kim / kpsf_im * (im_cen_phase / psf_im_cen_phase)
+        #
+        # The phases are complex exponentials
+        #
+        #  exp(ik*cen)
+        #
+        # So we can compute one phase as
+        #
+        #  im_cen_phase / psf_im_cen_phase = exp(ik * (im_cen - psf_cen))
+        #
+        # and then multiply it into the image.
+        #
+        # This operation and the deconvolutiomn will be done
+        # later in _measure_moments_fft
+
+        # now build the kernels
+        kres = _ksigma_kernels(
```

Comment:
maybe call this "kernels" to be consistent with the moments measuring code

Reply:
done.